tests/test_viewers.py

```diff
@@ -0,0 +1,63 @@
+import pytest
+from aiida import engine, orm
+
+
+@pytest.mark.usefixtures("aiida_profile_clean")
+def test_several_data_viewers(bands_data_object, folder_data_object):
+    from aiidalab_widgets_base import viewer, viewers
+
+    v = viewer(orm.Int(1))
+
+    # No viewer for Int, so it should return the input
+    assert isinstance(v, orm.Int)
+
+    # DictViewer
+    v = viewer(orm.Dict(dict={"a": 1}))
+    assert isinstance(v, viewers.DictViewer)
+
+    # BandsDataViewer
+    v = viewer(bands_data_object)
+    assert isinstance(v, viewers.BandsDataViewer)
+
+    # FolderDataViewer
+    v = viewer(folder_data_object)
+    assert isinstance(v, viewers.FolderDataViewer)
+
+    # ProcessNodeViewer
+    from aiida.workflows.arithmetic.add_multiply import add_multiply
+
+    result, workfunction = engine.run_get_node(
```

Comment:
I see, the rabbitmq is needed here for running the aiida process. I'd try to avoid this in the test. Can you try using the `generate_calc_job_node` fixture and add inputs/output to test this widget?

Reply:
Please check the example in qe plugin, the exit status seems can be directly set.

https://github.com/aiidateam/aiida-quantumespresso/blob/146bc57d74b76a11d7205a0f8449ae39c7675110/tests/conftest.py#L736-L751